src/crashhandler_unix.cpp

```diff
@@ -58,15 +58,28 @@ namespace {
 
    void restoreSignalHandler(int signal_number) {
 #if !(defined(DISABLE_FATAL_SIGNALHANDLING))
-      struct sigaction action;
-      memset(&action, 0, sizeof (action)); //
-      sigemptyset(&action.sa_mask);
-      action.sa_handler = SIG_DFL; // take default action for the signal
-      sigaction(signal_number, &action, NULL);
+      auto old_action_it = gSavedSigActions.find(signal_number);
+      if (old_action_it == gSavedSigActions.end()) {
+         // No saved action, so do nothing.
+         return;
+      }
+      
+      if (sigaction(signal_number, &(old_action_it->second), nullptr) < 0) {
+         std::string signal_name;
+         auto signal_name_it = gSignals.find(signal_number);
+         if (signal_name_it == gSignals.end()) {
+            signal_name = signal_name_it->second;
```

Comment:
Great pull request . This addresses exactly the issue

The “end” handling looks wrong. Once it at the end it can’t access the second and it would crash.

I’ll try it out

Reply:
Testing that logic is easy, you could just break out the error handling logic into its own function

Reply:
Good catch. I meant `!=`.